naga/src/back/msl/writer.rs

```diff
@@ -4349,10 +4349,10 @@ impl<W: Write> Writer<W> {
                 let name = self.namer.call("unpackSint32");
                 writeln!(
                     self.out,
-                    "metal::int {name}(uint b0, \
-                                       uint b1, \
-                                       uint b2, \
-                                       uint b3) {{"
+                    "int {name}(uint b0, \
```

Comment:
Since MSL `int` matches the name and type of the C++ `int`, the `metal::` prefix is flagged as being unnecessary. I don't completely understand it, but we need to remove it to pass our MSL validation step.